spec/models/category_spec.rb

```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :model do
+  before :each do
+    @user = User.new(name: 'sam', email: 'sam@gmail.com')
+    @category = Category.new(name: 'cow', icon: 'cow.png', user_id: @user.id)
+  end
+
+  it 'name should be present' do
+    expect(@category.name).to eq('cow')
+  end
+
+  it 'icon shold be present' do
+    expect(@category.icon).to eq('cow.png')
+  end
```

Comment:
- [ ] Kindly add integration tests for all the most important components of the RoR application. Integration tests help in identifying bugs that arise when different components are combined, detecting issues with communication between modules, and validating the overall behaviour of your app.